app/src/main/java/ch/hikemate/app/ui/map/HikeDetailsScreen.kt

```diff
@@ -132,9 +134,22 @@ fun HikeDetailScreen(
 
   val selectedHike by hikesViewModel.selectedHike.collectAsState()
 
+  // Gets initialized here so that the LaunchedEffect has access to it. The value is only actually
+  // initialised and used in HikeDetailsContent
+  val mapViewState = remember { mutableStateOf<MapView?>(null) }
```

Comment:
Suggestion: Would that work to remove the optional and initialize it to `hikeDetailsMap` right away ? In the `LaunchedEffect` in order to have access to `selectedHike`.

Reply:
Hmmm. How would that work? If we declare the variable inside of the Launched Effect, then its not in the global scope, so we would need to do some kind of lateinit, or? Also, we need the selectedHike to initialize the mapView, and that may still be null at this point in the code. The hike is only null-checked in the LaunchedEffect, however we need the mapState to implement the LaunchedEffect. This is a very awkward circualr dependency, but if you can think of a cleaner way to fix it I'd definitely be open to, I don't particularly love this solution either....

Reply:
@rBergCode asked me to help on this particular topic since I refactored the code in the first place.

The problem here is that :

- `LaunchedEffect(selectedHike)` is in the top composable function (top-level)
- When the `selectedHike` changes, we should save the map's state before leaving the screen
- To save the map's state, you need access to the instance of the map
- The map is defined at a lower level

So what you did is you brought the `mapView` instance up in the composable hierarchy. While this works, we might be able to avoid making `mapView` nullable.

To do this, instead of editing the existing `LaunchedEffect`, perhaps you could create a new `LaunchedEffect(selectedHike)` in `HikeDetailsContent` directly, where the map instance is created. Once it is created, you can execute the LaunchedEffect without any issue, and no need for null checks.

Reply:
Do you mean to create a second LaunchedEffect or move the existing one down?

Reply:
Create a second one.

Reply:
But then the first one in the higher level will execute first, im pretty sure, and once the back navigation is triggered the second one will never get a change to execute.

Reply:
Could you try it out maybe? I don't think it will cause any particular issues, at least I hope not.

Reply:
I have and it did not work

Reply:
I tried it out on my side too and it indeed does not work, so let's keep this way of doing. Sorry for bothering you!